### ISSUE TITLE
Do we need store for all api

### ISSUE THREAD

**Issue by ajitsinghkaler** [ajitsinghkaler/devto-clone#23]
In this repo we are adding store for all API but do we need to add that. The data is not reused you can directly call the api and it should work. why add a store layer

**Comment by fyodorio** [comment]
Store is a kinda opinionated mental model. And not a bad one. It allows to separate concerns of actions (or updaters in this case, as you use Component Store), effects (which aren't just server events but also modal windows, error catching, logging, and other side stuff), and observable selectors, which can be reused widely (with memoization, that's important) instead of creating new subjects (or anything similar) in each API consuming component.

So if you like such approach (I mean, you started to use it from the beginning, hadn't you) it's great to stick with this solid, well-documented and well-tested solution. 

But it's definitely up to you (and/or other maintainers).

**Comment by ajitsinghkaler** [comment]
Yeah but as I'm using it I think I use the API one time. Te store is used for nothing. We need store if we are reusing data in some way but I'm fetching data once showing it once. So, I'm starting to feel I need not store it in anyway. Component store is a service with a subject. I would not have even used service with a subject for all this

**Comment by fyodorio** [comment]
Well that makes sense. Anyway, if you don't see any use of the stores in future and you feel much more comfortable without them, why not get rid of it. It's always the matter of personal preference. I personally agree with an opinion of a great guy [Armen Vardanyan](https://twitter.com/armandotrue) who says something like "if an application is large enough for Angular, it's large enough for NgRx" 😄

In this case it's not the question of size but rather a tool choice though.
